src/haystack/filter/parser.rs

```diff
@@ -78,16 +78,18 @@ impl<'a, R: Read> Parser<Lexer<Scanner<'a, R>>> {
                     }
                 }
                 // Is A
-                TokenValue::Value(Value::Symbol(symbol)) => Ok(Term::IsA(IsA {
-                    symbol: symbol.clone(),
-                })),
+                TokenValue::Value(Value::Symbol(symbol)) => {
+                    let term = Term::IsA(IsA { symbol: symbol.clone() });
+                    self.lexer.read().ok();

```

Comment:
@rracariu although I haven't done this here, I think it would be good for the parser to throw an error if it detects whether there are any tokens left after the parser finished. This would help catch any other parser errors that could occur in future.